### PR TITLE
feat(travis): add travis ci job

### PR DESCRIPTION
**Pull request by 100stacks**
* move build step to `TravisCI` job

### CODE REVIEW

**Comment by 100stacks** [comment]
🤦‍♂ ...this was the issue: http://stackoverflow.com/questions/33820638/ddg#33820642

# TL;DR

I am making changes through the browser (https://github.com/100stacks/100stacks.github.io), which did not set the proper permission on `script/build` or `scripts/build`, which resulted in an error.  This is the solution:

```
before_install:
- chmod +x script/build
```


## This issue

```
$ script/build
/home/travis/.travis/functions: line 104: script/build: Permission denied
The command "script/build" exited with 126.
```

I tried any number of experiments to find a solution, which is one of the awesome abilities with GitHub and TravisCI open source!!!  Treat this a lab where you are free (**_literally_** 😊 ) to try out different things...experiment with ideas.  The frustrations you may experience are a crucial part of the learning process.

**TIP:** Telling or showing someone else what you learned will help reinforce it for you.  

## The immediate solution
Added this based on this StackOverflow (SO) thread (ref: http://stackoverflow.com/questions/33820638/ddg#33820642)

```
before_install:
- chmod +x script/build
```

I added the above before my `TravisCI` `jobs`, so it now looks like this:

```
before_install:
- chmod +x script/build 
jobs:
  include:
    - stage: build
      script: script/build
```

The `+x` means to change permission to allow `execution` (running the script) from this folder.  

## One caveat...

`chmod +x` command is not permeant until it's modified, which requires downloading and modify this repo from file system (aka command line).  That's not such a bad thing, since this current setup only allows script execution in TravisCI Docker GCP containers.

**Comment by 100stacks** [comment]
# Running Test Script Output

`htmlproofer` is not locating assets.

```
$ script/tests
Configuration file: /home/travis/build/100stacks/100stacks.github.io/_config.yml
            Source: /home/travis/build/100stacks/100stacks.github.io
       Destination: /home/travis/build/100stacks/100stacks.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
Invalid theme folder: _sass
      Remote Theme: Using theme pmarsceill/just-the-docs
                    done in 2.826 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
Running ["ScriptCheck", "ImageCheck", "HtmlCheck", "LinkCheck"] on ["./_site"] on *.html... 
Checking 4 external links...
Ran on 13 files!
- ./_site/blog/2016-05-19-angular-mongo-MEAN-stack-application/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/blog-template/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/containers/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/create-a-bot/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/docker/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/flask-web-server/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/list-vs-generator/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/microservices/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/os-cpu/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/python-generators/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/sys/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/blog/webhooks/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  internal image ./static/img/banner-bg.jpg does not exist (line 240)
  *  internal image ./static/img/intro-bg.jpg does not exist (line 215)
  *  internal image ./static/img/portfolio-4.jpg does not exist (line 261)
  *  internal script ./static/javascript/bootstrap.min.js does not exist (line 355)
  *  internal script ./static/javascript/jquery.js does not exist (line 352)
  *  internally linking to ./static/css/bootstrap.min.css, which does not exist (line 15)
     <link href="./static/css/bootstrap.min.css" rel="stylesheet">
  *  internally linking to ./static/css/landing-page.css, which does not exist (line 18)
     <link href="./static/css/landing-page.css" rel="stylesheet">
  *  internally linking to ./static/font-awesome/css/font-awesome.min.css, which does not exist (line 21)
     <link href="./static/font-awesome/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
- ./_site/index.html
  *  124:36: ERROR: ID about already defined (line 124)
  *  image ./static/img/banner-bg.jpg does not have an alt attribute (line 240)
  *  image ./static/img/intro-bg.jpg does not have an alt attribute (line 215)
  *  image ./static/img/portfolio-4.jpg does not have an alt attribute (line 261)
  *  linking to internal hash # that does not exist (line 60)
     <a class="navbar-brand topnav" href="#">Portfolio Page</a>
  *  linking to internal hash # that does not exist (line 72)
     <a href="#">Blog</a>
  *  linking to internal hash # that does not exist (line 326)
     <a href="#">Home</a>
htmlproofer 3.11.1 | Error:  HTML-Proofer found 187 failures!
The command "script/tests" exited with 1.
cache.2
store build cache
Done. Your build exited with 1.
```

**Comment by 100stacks** [comment]
- Verified the `script/tests` are running.
- Remove running tests from PR #5, until another site iteration.

## Remove for now...
```
    - stage: tests
      name: "Running build tests..."
      script: script/tests
```
